src/biokbase/narrative/jobs/job.py

```diff
@@ -347,47 +355,63 @@ def state(self, force_refresh=False):
             state = self.query_ee2_state(self.job_id, init=False)
             self._update_state(state)
 
-        return self._internal_state(JOB_INIT_EXCLUDED_JOB_STATE_FIELDS)
+        return self._internal_state(exclude)
 
     def _internal_state(self, exclude=None):
         """Wrapper for self._acc_state"""
         state = copy.deepcopy(self._acc_state)
         self._trim_ee2_state(state, exclude)
         return state
 
-    def output_state(self, state=None) -> dict:
+    def output_state(self, state=None, no_refresh=False) -> dict:
         """
-        :param state: can be queried individually from ee2/cache with self.state(),
-            but sometimes want it to be queried in bulk from ee2 upstream
-        :return: dict, with structure
-
-        {
-            outputWidgetInfo: (if not finished, None, else...) job.get_viewer_params result
-            jobState: {
-                job_id: string,
-                status: string,
-                created: epoch ms,
-                updated: epoch ms,
-                queued: optional - epoch ms,
-                finished: optional - epoc ms,
-                terminated_code: optional - int,
-                tag: string (release, beta, dev),
-                parent_job_id: optional - string or null,
-                run_id: string,
-                cell_id: string,
-                errormsg: optional - string,
-                error (optional): {
-                    code: int,
-                    name: string,
-                    message: string (should be for the user to read),
-                    error: string, (likely a stacktrace)
-                },
-                error_code: optional - int
```

Comment:
This output state I didn't find particularly helpful. Just looking at the "optional" error stuff seemed like a violation of helpful documentation. I concluded a long time ago that the error output state would probably be never triggered, and it's kind of mutually exclusive with the regular output state

Reply:
The error output state is triggered by an app failing somehow, which can happen pretty easily with some apps.

Here's the job state from an App Sleep job that I configured to fail:
```json
{
  "status": "error",
  "updated": 1650382783214,
  "queued": 1650382769657,
  "batch_job": false,
  "child_jobs": [],
  "retry_ids": [],
  "retry_count": 0,
  "job_id": "625ed7b128c29d4fd84dcf3a",
  "batch_id": null,
  "created": 1650382769000,
  "running": 1650382777753,
  "finished": 1650382783129,
  "errormsg": "Job output contains an error",
  "error": {
    "code": -32000,
    "name": "Server error",
    "message": "'App woke up from its nap very cranky!'",
    "error": "Traceback (most recent call last):\n  File \"/kb/module/bin/../lib/NarrativeTest/NarrativeTestServer.py\", line 101, in _call_method\n    result = method(ctx, *params)\n  File \"/kb/module/lib/NarrativeTest/NarrativeTestImpl.py\", line 345, in app_sleep\n    raise RuntimeError('App woke up from its nap very cranky!')\nRuntimeError: App woke up from its nap very cranky!\n"
  },
  "error_code": 1,
  "job_output": {}
}
```